app/src/modules/network/network.c

```diff
@@ -25,6 +30,34 @@ LOG_MODULE_REGISTER(network, CONFIG_APP_NETWORK_LOG_LEVEL);
 static struct net_mgmt_event_callback l4_cb;
 static struct net_mgmt_event_callback conn_cb;
 
+/* Do an additional upload to memfault to cloud once we connect, this will happen
```

Comment:
why this additional upload?

Reply:
To send coredump data right after a connection has been established.